dev-docs/best-practices/ui.md

```diff
@@ -0,0 +1,68 @@
+# UI
+
+The Taskcluster UI is implemented in the `ui/` directory of this repository.
+When we have a good reason to not follow the best practices in the UI, we document why.
+
+## Contents
+* [Tables](#tables)
+* [Links](#links)
+* [Buttons](#buttons)
+* [Text](#text)
+* [Spacing](#spacing)
+* [Color](#color)
+* [Visual Hierarchy](#visual-hierarchy)
+* [Document Header Hierarchy](#document-header-hierarchy)
+* [Tracking Changes](#tracking-changes)
+
+## Tables
+* Avoid multi-line text in a table - too much information for the given space. Alternatives:
+  * A [Card](https://material-ui.com/components/cards/).
+  * A [Drawer](https://material-ui.com/components/drawers/) that opens up via a button click in the table.
+* Right-align numbers - it’s easier to scan a column of numbers if the decimal part starts in the same location.
+
+## Links
+* Calculate link URLs before the user clicks on them. By doing so, a user will be able to do things like open links in a
+new tab or have the ability to inspect link URLs prior to the clicking.
+
+## Buttons
+* Destructive actions (e.g., a delete operation) should have a confirmation step.
```

Comment:
Do our current components have confirmation steps everywhere?

Reply:
Our current components don't have this at the moment but they should. I filed https://github.com/taskcluster/taskcluster/issues/957 yesterday to start the process. I plan a tracker issue to track all of the components that require this type of change.

> I see this can affect the PR I currently have in flight

We would need to add this change to the worker manager views. Feel free to ignore this for your current PR though since we don't have this implemented on any of our views. The tracker issue can take care of it.